roles/install-bridge-cni/tasks/main.yml

```diff
@@ -0,0 +1,40 @@
+- name: Create cni directory
+  file:
+    path: /opt/cni/bin
+    state: directory
+
+- name: Download and setup cni plugins
+  unarchive:
+    src: "{{ cni_plugins_url }}/{{ cni_plugins_version }}/{{ cni_plugins_tarball }}"
+    dest: "/opt/cni/bin"
+    remote_src: yes
+    extra_opts:
+      - --strip-components=1
+  retries: 3
+  delay: 5
+
+- name: Find regular files in a directory
+  find:
+    paths: /opt/cni/bin
+    file_type: file
+  register: found_files
+
+- name: Delete files except specific ones
+  file:
+    path: "{{ item.path }}"
+    state: absent
+  loop: "{{ found_files.files }}"
+  when: item.path not in ['/opt/cni/bin/host-local', '/opt/cni/bin/bridge', '/opt/cni/bin/portmap', '/opt/cni/bin/loopback']
```

Comment:
You don't need to delete these files, will not harm if they are around, I'm trying to just remove unnecessary code :)